provision/kubernetes/deploy.go

```diff
@@ -802,7 +802,7 @@ type serviceManager struct {
 
 var _ servicecommon.ServiceManager = &serviceManager{}
 
-func (m *serviceManager) CleanupServices(ctx context.Context, a provision.App, deployedVersion int, preserveOldVersions bool) error {
+func (m *serviceManager) CleanupServices(ctx context.Context, a provision.App, deployedVersion int, preserveOldVersions bool, newVersionSpec servicecommon.ProcessSpec) error {
 	depGroups, err := deploymentsDataForApp(ctx, m.client, a)
```

Comment:
@arthurcgc could we avoid call of `CleanupServices` using stop ?, stop operation should be a atomic update in replicas field.